src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs

```diff
@@ -47,24 +52,35 @@ public class TestRequestSender : ITestRequestSender
         private string clientExitErrorMessage;
 
         // Set default to 1, if protocol version check does not happen
-        // that implies host is using version 1
+        // that implies host is using version 1.
         private int protocolVersion = 1;
 
-        private int highestSupportedVersion = 2;
+        private int highestSupportedVersion = 3;
+
         private TestHostConnectionInfo connectionInfo;
 
+        private ITestRuntimeProvider runtimeProvider;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="TestRequestSender"/> class.
         /// </summary>
         /// <param name="protocolConfig">Protocol configuration.</param>
-        /// <param name="connectionInfo">Transport layer to set up connection</param>
-        public TestRequestSender(ProtocolConfig protocolConfig, TestHostConnectionInfo connectionInfo)
-            : this(connectionInfo, JsonDataSerializer.Instance, protocolConfig, ClientProcessExitWaitTimeout)
+        /// <param name="runtimeProvider">The runtime provider.</param>
+        public TestRequestSender(ProtocolConfig protocolConfig, ITestRuntimeProvider runtimeProvider)
+            : this(
+                  runtimeProvider,
+                  null,
```

Comment:
>null [](start = 18, length = 4)

nit: consider a named parameter for this.

Reply:
Done.